activemount.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+type activeMount struct {
+	UsageCount int
+}
+
+// activateVolume checks if the volume that has been requested to be mounted (as in docker volume mounting)
+// actually requires to be mounted (as an overlay fs mount). For that purpose check if other containers
+// have already mounted the volume (by reading in `activemountsdir`). It is also possible that the volume
+// has already been been mounted by the same container (when doing a `docker cp` while the container is running),
+// in that case the file named with the `requestId` will contain the number of times the container has
+// been requested the volume to be mounted. That number will be increased each time `activateVolume` is
+// called and decreased on `deactivateVolume`.
+// Parameters:
+//
+//	requestName: Name of the volume to be mounted
+//	requestID: ID of the container requesting the mount
+//	activemountsdir: Folder where Docker-On-Top mounts are tracked.
+//
+// Return:
+//
+//	doMountFs: Returns true if the request requires the filesystem to be mounted, false if not.
+//	err: If the function encountered an error, the error itself, nil if everything went right.
+func (d *DockerOnTop) activateVolume(requestName string, requestId string, activemountsdir lockedFile) (bool, error) {
+	var doMountFs bool
+
+	_, readDirErr := activemountsdir.ReadDir(1) // Check if there are any files inside activemounts dir
+	if readDirErr == nil {
+		// There is something no need to mount the filesystem again
+		doMountFs = false
+	} else if errors.Is(readDirErr, io.EOF) {
+		// The directory is empty, mount the filesystem
+		doMountFs = true
+	} else {
+		return false, fmt.Errorf("failed to list activemounts/ %v", readDirErr)
+	}
+
+	var activeMountInfo activeMount
+	activemountFilePath := d.activemountsdir(requestName) + requestId
+	file, err := os.Open(activemountFilePath)
+
+	if err == nil {
+		// The file can exist from a previous mount when doing a docker cp on an already mounted container, no need to mount the filesystem again
+		payload, readErr := io.ReadAll(file)
+		closeErr := file.Close()
```

Comment:
I just noticed we open the file and then on success just read all contents of it and close it. Looks just like [`os.ReadFile`](https://pkg.go.dev/os#ReadFile). Can we use it here?

Fun fact: Go's standard library [ignores](https://github.com/golang/go/blob/86bec1ec198f2720c83bd232a72b800b4ea5a9f6/src/os/file.go#L792) the error returned by `file.Close()` inside `os.ReadFile`